bluesky_queueserver/manager/plan_queue_ops.py

```diff
@@ -1,47 +1,233 @@
+            return "plan_uid" in plan
+
+        plans_to_remove = []
+        for plan in pq:
+            if not verify_plan(plan):
+                plans_to_remove.append(plan)
+
+        for plan in plans_to_remove:
+            await self._remove_plan(plan, single=False)
+
+        # Clean running plan info also (on the development computer it may contain garbage)
+        plan = await self._get_running_plan_info()
+        if plan and not verify_plan(plan):
+            await self._clear_running_plan_info()
+
+    async def _delete_pool_entries(self):
         """
-        Delete pool entries used by RE Manager. This method is mostly for use in testing.
+        See ``self.delete_pool_entries`` method.
         """
         await self._r_pool.delete(self._name_running_plan)
         await self._r_pool.delete(self._name_plan_queue)
+        await self._r_pool.delete(self._name_plan_history)
+        self._uid_set_clear()
+
+    async def delete_pool_entries(self):
+        """
+        Delete pool entries used by RE Manager. This method is mostly intended for use in testing,
+        but may be used for other purposes if needed.
+        """
+        async with self._lock:
+            await self._delete_pool_entries()
+
+    def _verify_plan_type(self, plan):
+        """
+        Check that the plan is a dictionary.
+        """
+        if not isinstance(plan, dict):
+            raise TypeError(f"Parameter 'plan' should be a dictionary: '{plan}', (type '{type(plan)}')")
+
+    def _verify_plan(self, plan):
+        """
+        Verify that plan structure is valid enough to be put in the queue.
+        Current checks: plan is a dictionary, `plan_uid` key is present, Plan with the UID is not in
+        the queue or currently running.
+        """
+        self._verify_plan_type(plan)
+        # Verify plan UID
+        if "plan_uid" not in plan:
+            raise ValueError("Plan does not have UID.")
+        if self._is_uid_in_set(plan["plan_uid"]):
+            raise RuntimeError(f"Plan with UID {plan['plan_uid']} is already in the queue")
+
+    def _new_plan_uid(self):
+        """
+        Generate UID for a plan.
+        """
+        return str(uuid.uuid4())
+
+    def set_new_plan_uuid(self, plan):
+        """
+        Replaces Plan UID with a new one or creates a new UID.
+
+        Parameters
+        ----------
+        plan: dict
+            Dictionary of plan parameters. The dictionary may or may not have the key `plan_uid`
+
+        Returns
+        -------
+        dict
+            Plan with new UID.
+        """
+        self._verify_plan_type(plan)
+        plan["plan_uid"] = self._new_plan_uid()
+        return plan
+
+    # --------------------------------------------------------------------------
+    #                          Operations with UID set
+
+    def _uid_set_clear(self):
+        """
+        Clear `self._uid_set`.
+        """
+        self._uid_set = set()
```

Comment:
Why not `self._uid_set.clear()`? This avoids the class of bugs where some other part of the program might be holding a reference to a stale versino of `_uid_set`. Experience with this led us to clear rather than recreate all the caches in `RunEngine` and `Bundler`.

Also, I am not convinced that these custom private methods add value over simply calling the methods on the private set itself.

Reply:
I guess you are right. I will change it in the next PR.

Reply:
Actually the idea behind the private methods was that they provide some level of abstraction. If the UIDs are to be stored in some different way (e.g. in Redis), then only those private methods need to be rewritten, not the code using them. So I am not sure if I wouldn't need to put those methods back if I remove them. So for now I will put TODO next to those methods and if we don't need them they could be removed in the future. 
Creating an instance of the set instead of clearing it is not right, so I will change it.